symengine/tests/eval/test_eval_double.cpp

```diff
@@ -107,6 +107,8 @@ TEST_CASE("eval_double: eval_double", "[eval_double]")
         {SymEngine::atan2(r1, neg(r2)), 2.08867384922582},
         {mul(pi, mul(E, EulerGamma)), 4.92926836742289},
         {pow(mul(EulerGamma, r4), integer(8)), 4813.54354505117582},
+        {pow(E, real_double(0.2)), 1.22140275816017},
```

Comment:
This test checks that whatever the result of `pow(E, real_double(0.2))` can be evaluated. What this PR does is evaluating `pow(E, real_double(0.2))` automatically without the need of evaluation using `eval_double` and the like. Can you add a test to check that `pow(E, real_double(0.2))` is a `RealDouble` and that it's value is close to `1.22140275816017`